### PR TITLE
Adds a fill() 2D operator

### PR DESCRIPTION
**Pull request by nickzoic**
#4340 ... Adds a fill() 2D operator much like hull() which gets rid of all the holes in a 2D shape.

TODO:

- [x] Is it possible/desirable to implement a 3D fill()?
- [x] Check I haven't done anything stupid in C++ 
- [ ] Add to docs / examples.
- [x] Tests still need work to cover more cases & check for correctness.

### CODE REVIEW

**Comment by nickzoic** [comment]
I'm ticking off "is it possible to implement a 3D fill" because yeah it might be possible but it's out of my depth for the moment so it can go be its own bug.

Along which lines, "have I done something stupid in C++?" I could use some help with.

I've added some tests for things like nested holes and fixed a bug I found as a result.

**Comment by t-paul** [comment]
Nice thanks! Yes, a 3D version would be awesome, but it's better to keep it separate in any case. I'll try to review as soon as possible, very much lacking time to spare lately. The test cases already look great :+1:.

**Comment by nickzoic** [comment]
On Tue, 13 Sep 2022, at 18:04, Torsten Paul wrote:
> Nice thanks! Yes, a 3D version would be awesome, but it's better to keep it separate in any case. I'll try to review as soon as possible, very much lacking time to spare lately. The test cases already look great 👍.
> 

OK thanks!  I wanted to test with all the 2D primitives but thinking about it I'm a bit worried that the default font one won't be pixel-identical on all build platforms.

**Comment by t-paul** [comment]
Yes, fonts can be tricky in that regard, we even had an issue with different versions of the Liberation fonts at some point.

For that reason the test cases come with their [own dedicated font files](https://github.com/openscad/openscad/tree/822a8c457a50618b70f0c0657b40d6355972574d/tests/data/ttf) trying to minimize the risk of platform differences breaking things.
https://github.com/openscad/openscad/blob/822a8c457a50618b70f0c0657b40d6355972574d/tests/test_cmdline_tool.py#L276-L279

**Comment by nickzoic** [comment]
Oh that's great, you're way ahead of my last-minute-before-bed conjectures :-)

**Comment by kwikius** [comment]
@nickzoic Looks like a nice feature. Will try to review in more detail when I get time. Also used your pull request to figure out how to add tests. Thanks for that :)

File: src/geometry/GeometryEvaluator.cc [hunk 1]
```diff
@@ -342,6 +355,8 @@ Polygon2d *GeometryEvaluator::applyToChildren2D(const AbstractNode& node, OpenSC
     return applyMinkowski2D(node);
   } else if (op == OpenSCADOperator::HULL) {
     return applyHull2D(node);
+  } else if (op == OpenSCADOperator::FILL) {
```
Comment (kwikius) [inline src/geometry/GeometryEvaluator.cc:371]: Am wondering if this section should now be changed to a switch statement to match switch statement a little later in the function?

Comment (t-paul) [reply src/geometry/GeometryEvaluator.cc:371]: Yep, might be nice, but I also don't mind keeping the if/else.

**Review by t-paul — APPROVED** [review]
Thanks, nice new feature!

**Pull request merged**